src/aead/aead.rs

```diff
@@ -76,37 +76,24 @@ impl OpeningKey {
     pub fn algorithm(&self) -> &'static Algorithm { self.key.algorithm() }
 }
 
-/// Authenticates and decrypts (&ldquo;opens&rdquo;) data in place.
-///
-/// The input is `in_out[in_prefix_len..]`; i.e. the input is the part of
-/// `in_out` after the prefix. When `open_in_place` returns `Ok(out_len)`, the
-/// decrypted output is `in_out[..out_len]`; i.e. the output has been written
-/// over the top of the prefix and the input. To put it a different way, the
-/// output overwrites the input, shifted by `in_prefix_len` bytes. To have the
-/// output overwrite the input without shifting, pass 0 as `in_prefix_len`.
-/// (The input/output buffer is expressed this way because Rust's type system
-/// does not allow us to have two slices, one mutable and one immutable, that
-/// reference overlapping memory at the same time.)
+/// Authenticates and decrypts (&ldquo;opens&rdquo;) data in place. When
+/// `open_in_place` returns `Ok(plaintext)`, the decrypted output is
+/// `plaintext`, which is a slice of `in_out`.
 ///
 /// C analog: `EVP_AEAD_CTX_open`
 ///
 /// Go analog: [`AEAD.Open`](https://golang.org/pkg/crypto/cipher/#AEAD)
-pub fn open_in_place(key: &OpeningKey, nonce: &[u8], in_prefix_len: usize,
-                     in_out: &mut [u8], ad: &[u8])
-                     -> Result<usize, error::Unspecified> {
+pub fn open_in_place<'a>(key: &OpeningKey, nonce: &[u8], ad: &[u8],
+                         in_out: &'a mut [u8])
```

Comment:
>  or make this one the more descriptive ciphertext_in_plaintext_out?

It would actually be `ciphertext_and_tag_in_plaintext_and_tag_out`. Totally doable.